lib/Driver/ToolChains.cpp

```diff
@@ -1530,20 +1553,16 @@ bool toolchains::GenericUnix::shouldProvideRPathToLinker() const {
 
 std::string toolchains::GenericUnix::getPreInputObjectPath(
     StringRef RuntimeLibraryPath) const {
-  // On Linux and FreeBSD and Haiku (really, ELF binaries) we need to add objects
-  // to provide markers and size for the metadata sections.
+  // On Elf platforms we need to add sentinel objects to provide markers and
+  // size for the metadata sections.
   SmallString<128> PreInputObjectPath = RuntimeLibraryPath;
-  llvm::sys::path::append(PreInputObjectPath,
-      swift::getMajorArchitectureName(getTriple()));
```

Comment:
Im working on removing the sentinels from the ELFish platforms in favor of the ELF linker tables.  Im hoping that I can figure out the one last issue and get that merged soon.

Reply:
awesome!

Reply:
FYI, I've gotten it mostly passing with the changes.  It uncovered a latent issue with the swift reflection dump tool that needs to be accounted for, but the rest of it is working now.  I think that testing with that is probably a good idea.

Reply:
Yeah, I'll pull it in locally and see how well it works after the 🦃 coma is over. Happy Giving Of Thanks!